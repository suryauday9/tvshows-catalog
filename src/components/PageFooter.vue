<template>
  <v-footer color="blue-grey darken-3">
    <div id="nav" class="nav-links">
      <router-link to="/">Home</router-link>
    </div>
    <v-col class="text-right white--text">
      <strong>copyright &copy; 2021 </strong>
    </v-col>
  </v-footer>
</template>

<script>
export default {
  name: "PageFooter"
};
</script>
<style scoped>
#nav a {
  color: #1976d2;
}
</style>
