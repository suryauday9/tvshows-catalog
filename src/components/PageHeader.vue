<template>
  <div class="home">
    <v-app-bar app color="blue-grey darken-3" dark flat>
      <router-link :to="{ name: 'showsCatalog' }" style="text-decoration: none">
        <v-btn text large>
          <v-icon x-large>mdi-youtube-tv</v-icon>
          <v-toolbar-title class="headline mx-5 display-1 font-weight-bold">
            <span class="display-1 text-uppercase font-weight-medium">TV </span>
            <span class="display-1 text-uppercase font-weight-light">Shows</span>
          </v-toolbar-title>
        </v-btn>
      </router-link>
      <v-spacer></v-spacer>
      <v-btn id='searchBar' v-if="$route.name === 'showsCatalog'" icon @click="showSearchbar()">
        <v-icon>mdi-magnify</v-icon>
      </v-btn>
    </v-app-bar>
  </div>
</template>

<script>
export default {
  name: "PageHeader",
  data() {
    return {
      drawer: false
    };
  },
  methods: {
    showSearchbar() {
      this.$store.dispatch('getSearchbarStatus');
    }
  }
};
</script>
