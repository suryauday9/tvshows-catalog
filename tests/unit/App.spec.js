import { shallowMount } from "@vue/test-utils";
import Vue from "vue";
import VueRouter from "vue-router";
import Vuetify from "vuetify";
import App from "../../src/App.vue";
import PageHeader from "../../src/components/PageHeader.vue";
import PageFooter from "../../src/components/PageFooter.vue";

describe("App component", () => {
  let appWrapper;
  const router = new VueRouter({ path: "/", name: "showsCatalog" });

  beforeEach(() => {
    Vue.use(VueRouter);
    Vue.use(Vuetify);
    appWrapper = shallowMount(App, {
      Vue,
      router
    });
  });

  afterEach(() => {
    appWrapper.destroy();
  });

  it('renders component correctly', () => {
    // see if component renders
    expect(appWrapper.element).toMatchSnapshot();
  });

  it("is a Vue instance", () => {
    expect(appWrapper.isVueInstance).toBeTruthy();
  });

  it("it should have a v-app-stub", () => {
    expect(appWrapper.html()).toContain("v-app");
  });

  describe("it should load PageHeader component", () => {
    it("it should load the page-header", () => {
      expect(PageHeader).toBeTruthy();
    });
  });

  describe("it should load PageFooter component", () => {
    it("it should load the page-footer", () => {
      expect(PageFooter).toBeTruthy();
    });
  });
});
