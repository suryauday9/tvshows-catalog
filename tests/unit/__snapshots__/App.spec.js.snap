// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders component correctly 1`] = `
<v-app-stub
  id="app"
>
  <header
    class="sticky"
  >
    <page-header-stub />
  </header>
   
  <v-main-stub
    class="white darken-4"
    tag="main"
  >
    <transition-stub
      name="fade"
    >
      <router-view-stub
        name="default"
      />
    </transition-stub>
  </v-main-stub>
   
  <footer>
    <page-footer-stub />
  </footer>
</v-app-stub>
`;
